### PR TITLE
feat(input): add github API URL configuration for input

### PR DESCRIPTION
**Pull request by Meroje**
Hello again :) I was moving my workflow from a `go install` to this action (but you've since pushed 0.6.1 which would not require this) and found that because I'm running on a Github Enterprise appliance the default octokit client would not call github.com to find gflows' releases. (the relevant line is here https://github.com/actions/toolkit/blob/2f164000dcd42fb08287824a3bc3030dbed33687/packages/github/src/internal/utils.ts#L24)
I'm proposing to add an input to override the baseUrl option as the corresponding env variable can not be changed.

Still debating wether it should force github.com as the default though.

### CODE REVIEW

**Comment by jbrunton** [comment]
Oh, interesting – I didn't know about this behaviour.

@Meroje: Are there reasons to let users override the base URL which I'm missing, however? I would have assumed all users, wherever they run this action, would want the API used by the action to be `https://api.github.com` (since that's where gflows is hosted). In which case, it might be better to just specify that as the `baseUrl` in the octokit config, and then there's no need for the extra input option. What do you think?

**Comment by Meroje** [comment]
This is exactly what I'm pondering on, we do copy actions we use to our appliance to lower the dependency on github.com so there would even be an argument for overriding event the repo owner/name combo, but I agree the most wanted default is to use api.github.com.

**Comment by jbrunton** [comment]
To confirm:

1. I assume that this action fails on GitHub Enterprise installations? (Do you get a not found error of some kind?)
2. It sounds like you would still like the option to override the API url if you copy the action to your Enterprise installation?

If so, I think it might make sense to default to api.github.com, since then at least the action works in all cases. To solve for 2) you could keep the input option, and I think override it something like this:

```yaml
steps:
- uses: jbrunton/setup-gflows@v1
  with:
    token: ${{ secrets.PUB_GITHUB_TOKEN }}
    github-api-url: ${{ env.GITHUB_API_URL }}
```

**Comment by Meroje** [comment]
1. Yes that's what happened 
image
2. Not sure I'd do that, we're only syncing actions at this point (using https://github.com/actions/actions-sync), as long as the tool cache can be used in absence of an answer from github.com

File: src/main.ts [hunk 1]
```diff
@@ -7,15 +7,20 @@ import * as core from '@actions/core'
 import * as github from '@actions/github'
 import {GitHub} from '@actions/github/lib/utils'
 
+const GITHUB_API_URL = 'https://api.github.com'
+
 function createOctokit(): Octokit {
   const token = core.getInput('token')
+  const gitHubOptions = {
+    baseUrl: core.getInput('github-api-url') || GITHUB_API_URL
```
Comment (jbrunton) [inline src/main.ts]: Consider this optional, but can we specify a default in the action.yml file instead? It avoids the need for the extra const and conditional.

File: action.yml [hunk 2]
```diff
@@ -13,6 +13,10 @@ inputs:
     description: "The version of GFlows to install (default: latest)"
     required: false
     default: latest
+  github-api-url:
+    description: 'configure github API URL. Default `$GITHUB_API_URL`'
```
Comment (jbrunton) [inline action.yml]: If I'm reading your latest commit correctly, I think this description should be updated?

```suggestion
    description: 'configure github API URL. Default `https://api.github.com`'
```

**Comment by jbrunton** [comment]
@Meroje: Thanks for clarifying!

If I read your changes correctly (it's late here!) it looks like you amended the PR to default to api.github.com? I left a couple of suggestions if so. I don't mind too much about the conditional, but the description needs updating. Thanks!

File: action.yml (hunk 2 above)
Comment (Meroje) [reply action.yml]: oh yea absolutely :)

**Comment by Meroje** [comment]
I did change the default as you suggested, thanks for the review

**Comment by jbrunton** [comment]
Thanks! A few build failures which I think are unrelated to your changes. I'll take a look and see what I need to fix.

**Comment by jbrunton** [comment]
@Meroje: Sorry for the delay getting back to you. Would you be kind enough to rebase with latest develop? I think that should address the build errors.

**Comment by jbrunton** [comment]
@Meroje: merged and released. Thanks for the PR!

**Pull request merged**